bin/build

```diff
@@ -33,8 +31,11 @@ cp -a "$BP_DIR/middleware/." $MW_LAYER
 local_lock_checksum=$(sha256sum "${MW_LAYER}/package-lock.json" | cut -d " " -f 1)
 cached_lock_checksum=$(yj -t < "${NODE_MODULES_DIR}.toml" | jq -r ".metadata.package_lock_checksum")
 
-if [[ "$local_lock_checksum" == "$cached_lock_checksum" ]] ; then
-    echo "using previous node_modules and dist artifacts from cache"
+local_stack=${CNB_STACK_ID}
+cached_stack=$(yj -t < "${NODE_MODULES_DIR}.toml" | jq -r ".metadata.stack")
```

Comment:
do we need to default this to "heroku-18" or it will break the cache of existing builds on next push?

Reply:
Yeah, we could. I didn't think it was worth it since we are only in pilot status.

Reply:
that's fair. I forget how expensive / how much of a pain it is.